src/query/block/scalar.go

```diff
@@ -57,23 +57,27 @@ func (b *Scalar) StepIter() (StepIter, error) {
 	bounds := b.meta.Bounds
 	steps := bounds.Steps()
 	return &scalarStepIter{
-		meta: b.meta,
-		step: scalarStep{
-			vals: []float64{b.val},
-		},
+		meta:    b.meta,
+		s:       b.s,
 		numVals: steps,
 		idx:     -1,
 	}, nil
 }
 
+// ScalarFunc determines the function to apply to generate the value at each step
+type ScalarFunc func(t time.Time) float64
```

Comment:
This seems really hacky that the base op needs a function that takes in time even when it doesn't need to. Can you localize the changes required rather than adding functions that necessarily take unnecessary params?